Added readMe (#12)
KostiantynVorobiov/My-own-project-Internet-shop-: README.md

> My own project "Internet shop" 

Reviewer: ```suggestion
# My own project "Internet shop" 
```
The # makes the text look like a headline :)

> Project purpose.
> Create a simple online store selling mobile phones with the ability to register new users and add new products to the store's catalog. 
> Restricting access to pages depending on the role, there are two of them user and admin. 
> Saving all information about the user and products in the database. 
> The user selects products, adds them to the cart and places an order to purchase the selected products. 
> The admin can view information about all users, and the product with the ability to edit and delete them as needed.

Reviewer: ```suggestion
## Project purpose.
To create a simple online store selling mobile phones with the ability to register new users and add new products to the store's catalog. 
Access to pages  is restricted depending on the website visitor's role; there are two roles: user and admin. 
All information about the user and products is saved in the database. 
The user selects products, adds them to the cart and places an order to purchase the selected products. 
The admin can view information about all users and the product with the ability to edit and delete them as needed.
```

> Project structure.
> Project have N-tier architecture.
> There is a Dao level where the operations of accessing the storage are described, at the first stage these were lists, but over time they went over and implemented the database.
> There is a service level where the business logic of the project is implemented.
> There is a controller layer that is responsible for receiving requests from users, processing and sending responses back.
> And the level where the models of our store are located.

Reviewer: ```suggestion
## Project structure.
The project has N-tier architecture.
There is a Dao level which defines the operations for accessing the storage. At first, the storage was implemented as lists, and, at a later stage in project development, access to a MySQL database was implemented.
There is a service level where the business logic of the project is implemented.
There is a controller layer that is responsible for receiving requests from users, processing and sending responses back.
```
I think you can add info about the _presentation layer_, which we implemented with JSP's)

> Implementation details.
> The project was created on Java in IntelliJ IDEA in compliance with the SOLID principles. 
> When creating, the following technologies were used: TomCat as a server, 
> JSP for displaying pages with information, JDBC for executing queries to the database, 
> Workbench MySQL for working with a database and Maven for assembling a project.

Reviewer: ```suggestion
## Implementation details.
The project was created on Java in IntelliJ IDEA in compliance with the SOLID principles. 
The technologies used in the project are as follows: 
- TomCat as the server;
- JSP for displaying web pages;
- JDBC for executing queries to the database; 
- MySQL Workbench for work with the database;
- Maven for assembling the project.
```

> Launch guide.
> Clone project from this repository.
> Download and install TomCat server and add all the necessary dependencies to the pom.xml file required for Java servlets.
> Install Workbench MySQL, install the necessary driver to connect to your database and change the driver in the class, your username and password, also specify the url address for connecting to the database. 
> Create a schema in your database and using the queries that you will find in the resourses folder in the init_db file, queries to create all the necessary tables.
> Run the application, first register as a new user and log in using the data you entered during registration. 
> When you enter the site, click on the inject, this will load the information for testing the functionality into the database.
> Test all accessible pages for the user. 
> To stream the admin functionality, you need to log out of the system and log in with the login and password for the admin (login: admin, password: a), he loaded by pressing the inject button.

Reviewer: ```suggestion
## Launch guide.
Clone the project from this repository.
Download and install the Tomcat server and add all the necessary dependencies for Java servlets to the `pom.xml` file.
Install MySQL Workbench, install the necessary driver to connect to your database and change the driver in the class, insert your username and password, and specify the url address for connecting to the database. 
Create a schema in your database using the queries that you will find in the resources folder in the `init_db` file.
When you run the application, first register as a new user and log in using the data you entered during registration. 
When you log on to the site, click on the `Inject` button, which will load the data for testing the website functionality into the database.
Test all pages accessible for the user. 
To try out the admin functionality, you need to log out of the system and log in with the admin login and password (login: _admin_, password: _a_). The admin profile will have been created when you clicked on the inject button.
```
I think you need to specify which class or file you're talking about here:  `change the driver in the class, insert your username and password, and specify the url address for connecting to the database.`

> Author.
> Kostiantyn Vorobiov
> https://github.com/KostiantynVorobiov

Reviewer: ```suggestion
## Author.
[Kostiantyn Vorobiov](https://github.com/KostiantynVorobiov)
```

Author: Done

The user selects products, adds them to the cart and places an order to purchase the selected products. 
The admin can view information about all users and the product with the ability to edit and delete them as needed.

## Project structure.
The project has N-tier architecture.
There is a Dao level which defines the operations for accessing the storage. At first, the storage was implemented as lists, and, at a later stage in project development, access to a MySQL database was implemented.
There is a service level where the business logic of the project is implemented.
There is a controller layer that is responsible for receiving requests from users, processing and sending responses back.
The presentation layer, which we implemented with JSP.

## Implementation details.
The project was created on Java in IntelliJ IDEA in compliance with the SOLID principles. 
The technologies used in the project are as follows: 
- TomCat as the server;
- JSP for displaying web pages;
- JDBC for executing queries to the database; 
- MySQL Workbench for work with the database;
- Maven for assembling the project.

## Launch guide.

> Clone the project from this repository.
> Download and install the Tomcat server and add all the necessary dependencies for Java servlets to the `pom.xml` file.
> Install MySQL Workbench, in class `ConnectionUtil` install the necessary driver to connect to your database and change the driver in the class, insert your username and password, and specify the url address for connecting to the database. 
> Create a schema in your database using the queries that you will find in the resources folder in the `init_db` file.
> When you run the application, first register as a new user and log in using the data you entered during registration. 
> When you log on to the site, click on the `Inject` button, which will load the data for testing the website functionality into the database.
> Test all pages accessible for the user. 
> To try out the admin functionality, you need to log out of the system and log in with the admin login and password (login: _admin_, password: _a_). The admin profile will have been created when you clicked on the inject button.

Reviewer 2: ```suggestion
1. Clone the project from this repository.
2. Download and install the Tomcat server and add all the necessary dependencies for Java servlets to the `pom.xml` file.
3. Install MySQL Workbench, in class `ConnectionUtil` install the necessary driver to connect to your database and change the driver in the class, insert your username and password, and specify the url address for connecting to the database. 
4. Create a schema in your database using the queries that you will find in the resources folder in the `init_db` file.
When you run the application, first register as a new user and log in using the data you entered during registration. 
When you log on to the site, click on the `Inject` button, which will load the data for testing the website functionality into the database.
5. Test all pages accessible for the user. 
6. To try out the admin functionality, you need to log out of the system and log in with the admin login and password (login: _admin_, password: _a_). The admin profile will have been created when you clicked on the inject button.
```

Author: Done
